core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -0,0 +1,190 @@
+            telemetryClient.track(requestFailureCountSt);
+        }
+
+        double durationAvg = getRequestDurationAvg(local);
+        if (durationAvg != 0) {
+            MetricTelemetry requestDurationSt = createStatsbeatTelemetry(REQUEST_DURATION, durationAvg);
+            requestDurationSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(requestDurationSt);
+        }
+
+        if (local.retryCount.get() != 0) {
+            MetricTelemetry retryCountSt = createStatsbeatTelemetry(RETRY_COUNT, local.retryCount.get());
+            retryCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(retryCountSt);
+        }
+
+        if (local.throttlingCount.get() != 0) {
+            MetricTelemetry throttleCountSt = createStatsbeatTelemetry(THROTTLE_COUNT, local.throttlingCount.get());
+            throttleCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(throttleCountSt);
+        }
+
+        if (local.exceptionCount.get() != 0) {
+            MetricTelemetry exceptionCountSt = createStatsbeatTelemetry(EXCEPTION_COUNT, local.exceptionCount.get());
+            exceptionCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(exceptionCountSt);
+        }
+    }
+
+    // TODO create a new instrumentation set automatically when its capacity reaches 64
+    public void addInstrumentation(String instrumentation) {
+        synchronized (lock) {
+            current.instrumentationList.add(instrumentation);
+        }
+    }
+
+    public Set<String> getInstrumentationList() {
+        return current.instrumentationList;
+    }
+
+    /**
+     * @return a 64-bit long that represents a list of instrumentations enabled. Each bitfield maps to an instrumentation.
+     */
+    public long getInstrumentation() {
+        return StatsbeatHelper.encodeInstrumentations(current.instrumentationList);
+    }
+
+    public void incrementRequestSuccessCount() {
+        current.requestSuccessCount.incrementAndGet();
+    }
+
+    public void incrementRequestFailureCount() {
+        current.requestFailureCount.incrementAndGet();
+    }
+
+    public void addRequestDuration(double duration) {
+        synchronized (lock) {
+            current.requestDurationCount.incrementAndGet();
+            current.totalRequestDuration.getAndAdd(duration);
+        }
+    }
+
+    public void incrementRetryCount() {
+        current.retryCount.incrementAndGet();
+    }
+
+    public void incrementThrottlingCount() {
+        current.throttlingCount.incrementAndGet();
+    }
+
+    public void incrementExceptionCount() {
+        current.exceptionCount.incrementAndGet();
+    }
+
+    public long getRequestSuccessCount() {
```

Comment:
can you mark the methods that are only used by tests (and make package protected if possible)?